File: 1422. Maximum Score After Splitting a String.py
class Solution:
    def maxScore(self, s: str) -> int:
        ans, score = 0, s.count('1')
        
        for i in range(len(s)-1):
            if s[i]=='0':
                score += 1
            
            else:
                score -= 1
            
            ans = max(ans, score)
        
        return ans

File: 1422. Maximum Score After Splitting a String.js
/**
 * @param {string} s
 * @return {number}
 */
 var maxScore = function(s) {
    let ans = score = 0;

    for(let c of s){
        if(c=='1'){
            score += 1;
        }
    }
    for(let i=0;i<s.length-1;i++){
        if(s[i]=='1'){
            score -= 1;
        }else{
            score += 1;
        }
        ans = Math.max(ans, score);
    }
    return ans;
};